### ISSUE TITLE
How if I want to continue the training from your finetuned checkpoint?

### ISSUE THREAD

**Issue by LiuYixian** [microsoft/MASS#102]
I continue training with the options:

> model/cnndm_evaluation/processed_test/
> --user-dir mass
> --task translation_mass
> --arch transformer_mass_base
> --optimizer adam
> --adam-betas '(0.9,0.98)'
> --clip-norm 0.0
> --lr 0.0005
> --min-lr 1e-09
> --lr-scheduler inverse_sqrt
> --weight-decay 0.0
> --criterion label_smoothed_cross_entropy
> --label-smoothing 0.1
> --update-freq 8
> --max-tokens 4096
> --ddp-backend no_c10d
> --max-epoch 25
> --max-source-positions 512
> --max-target-positions 512
> --skip-invalid-size-inputs-valid-test
> --restore-file model/cnndm_evaluation/checkpoint.pt
> --save-dir modified_ex7_weight_0.8_0
> --warmup-init-lr 1e-7
> --warmup-updates 4000
But the result is bad. 

> | model/cnndm_evaluation/processed/ train src-tgt 287113 examples
> | epoch 015:   1000 / 4439 loss=3.868, nll_loss=2.178, ppl=4.53, wps=248, ups=0, wpb=4052.054, bsz=64.771, num_updates=446, lr=5.58389e-05, gnorm=1.407, clip=0.000, oom=0.000, wall=7294, trai
> | epoch 015:   2000 / 4439 loss=3.880, nll_loss=2.192, ppl=4.57, wps=248, ups=0, wpb=4024.655, bsz=64.631, num_updates=1446, lr=0.000180814, gnorm=1.457, clip=0.000, oom=0.000, wall=23486, tr
> | epoch 015:   3000 / 4439 loss=3.962, nll_loss=2.287, ppl=4.88, wps=267, ups=0, wpb=4019.826, bsz=64.729, num_updates=2446, lr=0.000305789, gnorm=1.520, clip=0.000, oom=0.000, wall=36799, tr
> | epoch 015:   4000 / 4439 loss=4.093, nll_loss=2.438, ppl=5.42, wps=280, ups=0, wpb=4018.712, bsz=64.713, num_updates=3446, lr=                                                               0.000430764, gnorm=1.608, clip=0.000, oom=0.000, wall=49503, train_wall=24948
> | epoch 015 | loss 4.164 | nll_loss 2.520 | ppl 5.73 | wps 284 | ups 0 | wpb 4022.561 | bsz 64.692 | num_updates 3884 | lr 0.000                                                               485503 | gnorm 1.734 | clip 0.000 | oom 0.000 | wall 54934 | train_wall 26075
> | epoch 015 | valid on 'valid' subset | loss 4.867 | nll_loss 3.287 | ppl 9.76 | num_updates 3884 | best_loss 4.54276
> | saved checkpoint modified_ex1_weight_0.3_single_p40_0/checkpoint15.pt (epoch 15 @ 3884 updates) (writing took 7.896648406982422 seconds)
> | epoch 016:   1000 / 4439 loss=4.782, nll_loss=3.227, ppl=9.36, wps=327, ups=0, wpb=4012.957, bsz=64.855, num_updates=4885, lr=0.000452447, gnorm=2.821, clip=0.000, oom=0.000, wall=67413, train_wall=28657
> | epoch 016:   2000 / 4439 loss=4.798, nll_loss=3.248, ppl=9.50, wps=326, ups=0, wpb=4010.803, bsz=64.752, num_updates=5885, lr=0.000412218, gnorm=2.299, clip=0.000, oom=0.000, wall=79736, train_wall=31232
> | epoch 016:   3000 / 4439 loss=4.812, nll_loss=3.265, ppl=9.61, wps=324, ups=0, wpb=4019.961, bsz=64.725, num_updates=6885, lr=0.000381108, gnorm=2.187, clip=0.000, oom=0.000, wall=92347, train_wall=33806
> | epoch 016:   4000 / 4439 loss=4.813, nll_loss=3.268, ppl=9.63, wps=324, ups=0, wpb=4021.669, bsz=64.694, num_updates=7885, lr=0.000356122, gnorm=2.114, clip=0.000, oom=0.000, wall=104839, train_wall=36379
> | epoch 016 | loss 4.811 | nll_loss 3.267 | ppl 9.63 | wps 324 | ups 0 | wpb 4020.482 | bsz 64.680 | num_updates 8323 | lr 0.000346625 | gnorm 2.065 | clip 0.000 | oom 0.000 | wall 110258 | train_wall 37503
> | epoch 016 | valid on 'valid' subset | loss 4.778 | nll_loss 3.181 | ppl 9.07 | num_updates 8323 | best_loss 4.54276
> | saved checkpoint modified_ex1_weight_0.3_single_p40_0/checkpoint16.pt (epoch 16 @ 8323 updates) (writing took 8.073232889175415 seconds)
> | done training in 110436.7 seconds

The loss is increasing during continued training. I think something like learning rate is wrong in the continued training.

Could you tell me what to do if I want to continue training with your checkpoint? Thank you.

**Comment by StillKeepTry** [comment]
It seems that you are using warmup learning rate. It is suitable to train model from scratch, but not to continue train the fine-tuned checkpoint. I think you should decrease your warmup steps and learning rate.
